numpy/_core/tests/test_ufunc.py

```diff
@@ -2972,6 +2972,13 @@ def test_resolve_dtypes_basic(self):
         with pytest.raises(TypeError):
             np.add.resolve_dtypes((i4, f4, None), casting="no")
 
+    def test_resolve_dtypes_comparison(self):
```

Comment:
This test failed if one used the old `PyIntAbstractDType` directly - see https://github.com/numpy/numpy/pull/26116#issuecomment-2016508044